Add the lessons_progresses and all(_,raw) methods

herminiotorres opened the pull request:
Add these methods to Edools::Enrollment:
- every
- lessons_progresses

Add usage example to README

Add tests for both methods:
- every
- lessons_progresses

herminiotorres committed:
Add the lessons_progresses and every methods

Add these methods to Edools::Enrollment:
- every
- lessons_progresses

Add usage example to README

Add tests for both methods:
- every
- lessons_progresses

regismesquita commented on lib/edools/version.rb:
```diff
@@ -1,5 +1,5 @@
 
 module Edools
-  VERSION = '0.2.0'
+  VERSION = '0.3.0'
```
como não tem grandes mudanças achava melhor fazer o bump menor (i.e. 0.2.1)


-- EDIT

Um hotfix foi lançado com a 0.2.1 então tomei a liberdade de mudar a desse MR para 0.2.2

Commit:
Merge branch 'master' into feature/add_all_without_response_filter_and_lessons_progresses

regismesquita commented on lib/edools/enrollment.rb:
```diff
@@ -46,5 +46,16 @@ def self.update(data = {})
       new exception.errors
     end
+
+    def self.every(data = {})
```
This name is not clear enough , forcing us to read the documentation , neither the documentation is clear , what do you mean by "Entire query data" the entire set without pagination? unstructured data? structured data with every field?

What is exactly is the difference between the return of `all` and `every` , also every gives us the idea of a loop , is it an enumerable?

herminiotorres replied:
I did not find a better name to represent the method.

So, the response in all method is handled for the attributes set of the Edools::Enrollment class.

where are they:
- id
- first_name
- email
- type
- errors

IMHO, I don't like this.

But not to break anything that depends on the all method, I created the every method.

if you have a better suggestion... I'm listening 🗣👂

regismesquita replied:
Since it's doing the same job os the all method , it would be better to allow an extra parameter for the `all` method , allowing it to return the results unstructured.

herminiotorres committed:
Remove the every method

Change the all method. The all method can display the response raw data or not.
